pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -179,6 +179,15 @@ if (( "${NIX_DEBUG:-0}" >= 1 )); then
 fi
 
 PATH="$path_backup"
+
+# Optionally run external hook.
+if [ -n "${NIX_CC_WRAPPER_HOOK:-}" ]; then
+    "$NIX_CC_WRAPPER_HOOK" @prog@ \
+        ${extraBefore+"${extraBefore[@]}"} \
+        ${params+"${params[@]}"} \
+        ${extraAfter+"${extraAfter[@]}"}
+fi
```

Comment:
Should compilation still happen when you provide this hook?

Reply:
Not for my hook, make would probably fail if nothing is actually built. But maybe there's some other usecase.